### ISSUE TITLE
PropertyNotSetInConstructor: Property Doctrine\ORM\EntityRepository::$_entityName is not defined in constructor of Repository and in any methods called in the constructor

### ISSUE THREAD

**Issue by ruudk** [weirdan/doctrine-psalm-plugin#30]
Thanks for this plugin. It removes a lot of false errors from my codebase ❤️ 

I wonder if we can also fix this error:
 ```
INFO: PropertyNotSetInConstructor - src/Infrastructure/Persistence/Doctrine/Repository/DoctrineSettlementRepository.php:15:13 - Property Doctrine\ORM\EntityRepository::$_entityName is not defined in constructor of Infrastructure\Persistence\Doctrine\Repository\DoctrineSettlementRepository and in any methods called in the constructor
final class DoctrineSettlementRepository extends AbstractServiceEntityRepository implements SettlementRepository
```

code:
```php
final class DoctrineSettlementRepository extends AbstractServiceEntityRepository implements SettlementRepository
{
    public function __construct(RegistryInterface $registry)
    {
        parent::__construct($registry, Settlement::class);
    }
```

**Comment by ruudk** [comment]
@weirdan do you have tips how to approach this? Maybe I can create a PR to solve it.

**Comment by ruudk** [comment]
This solves it: https://github.com/weirdan/doctrine-psalm-plugin/pull/31

**Comment by weirdan** [comment]
@ruudk do you have a self-contained reproducer for this (one that does not rely on your internal classes)? I've seen this error before, but never managed to succinctly reproduce it. As far as I recon it's a Psalm bug, and a way to reproduce would be the first step to fix it there.

**Comment by ruudk** [comment]
```php
use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
use Symfony\Bridge\Doctrine\RegistryInterface;

/**
 * @template T
 * @template-extends ServiceEntityRepository<T>
 */
abstract class AbstractServiceEntityRepository extends ServiceEntityRepository
{
    /**
     * @var RegistryInterface
     */
    private $registry;

    public function __construct(RegistryInterface $registry, string $entityClass)
    {
        parent::__construct($registry, $entityClass);

        $this->registry = $registry;
    }
}

class Settlement {}

final class DoctrineSettlementRepository extends AbstractServiceEntityRepository
{
    public function __construct(RegistryInterface $registry)
    {
        parent::__construct($registry, Settlement::class);
    }
}
```

Does this work?

**Comment by weirdan** [comment]
Reproduced with https://github.com/weirdan/psalm-bug-PropertyNotSetInConstructor-with-stub

**Comment by ruudk** [comment]
Thanks
